R/log_shiny_input_changes.R

```diff
@@ -0,0 +1,93 @@
+#' Auto logging input changes in Shiny app
+#'
+#' This is to be called in the \code{server} section of the Shiny app.
+#'
+#' Function taken from \pkg{logger} package and improved in this PR
+#' [daroczig/logger/pull/155](https://github.com/daroczig/logger/pull/155/). Once this PR gets merged, the function will
+#' be removed.
+#'
+#' @keywords internal
+#'
+#' @param input passed from Shiny's \code{server}
+#' @param level log level
+#' @param excluded_inputs character vector of input names to exclude from logging
+#' @param excluded_patterns character of length one including a grep pattern of names to be excluded from logging
+#' @param skip_init should initialization message be displayed
+#' @param namespace the name of the namespace
+#' @importFrom utils assignInMyNamespace assignInNamespace
+#' @examples \dontrun{
+#' library(shiny)
+#'
+#' ui <- bootstrapPage(
+#'   numericInput("mean1", "mean1", 0),
+#'   numericInput("mean2", "mean2", 0),
+#'   numericInput("sd", "sd", 1),
+#'   textInput("title", "title", "title"),
+#'   textInput("foo", "This is not used at all, still gets logged", "foo"),
+#'   passwordInput("password", "Password not to be logged", "secret"),
+#'   plotOutput("plot")
+#' )
+#'
+#' server <- function(input, output) {
+#'   log_shiny_input_changes(input, excluded_inputs = "password", exclude_patterns = "mean")
+#'
+#'   output$plot <- renderPlot({
+#'     hist(rnorm(1e3, input$mean, input$sd), main = input$title)
+#'   })
+#' }
+#'
+#' shinyApp(ui = ui, server = server)
+#' }
+log_shiny_input_changes <- function(
+    input,
+    level = logger::INFO,
```

Comment:
```suggestion
    level = logger::TRACE,
```

Reply:
I tried to get this function to be as much similar to `logger::log_shiny_input_changes` as possible. 
This would be a helpful change, but I don't think this default value will be ever changed in logger.

Reply:
I was thinking that this `teal.logger::log_shiny_input_changes` function could stay even after the merge of your PR in `logger`. We can keep it as a higher level interface to either our internally built functionality or the one exported in `logger`.
In that case, we have full control on the function arguments as well as function body. We can even remove `level` at all. That actually would be even better. This would also allow us to make more changes and align it more to our needs (this is regarding the usage of `assignInMyNamespace` and `assignInNamespace`).

Reply:
Got it, so let me release the kraken and scrap the current implementation and just write the way it suits our needs the most